docs/developer-notes.md

```diff
@@ -76,6 +76,41 @@ Scuri uses [Jasmine's spy feature](https://jasmine.github.io/2.0/introduction.ht
 We loosely follow the [Jasmine Style Guide](https://github.com/CareMessagePlatform/jasmine-styleguide) when writing tests. 
 One main guideline is that a `describe` block should be created for each method / scenario under test, and an `it` block should be created for each property being verified.
 
+### E2E Testing
+#### CATcher Test Prep
+E2E Tests are currently run using [Protractor](http://www.protractortest.org/#/) testing framework with the following stages. 
+1. Build CATcher with using `test` architecture.
+   - Using `test` build configuration located in `angular.json` under `projects.catcher.architect.configurations` we build a version of CATcher within a test environment that replaces `src/environments/environment.ts` with `src/environments/environment.test.ts` on runtime. This allows for the feeding of information into the application to differ its actions in comparison to the default / production environments.
+2. Provide Test Environment Information
+   - The Test Environment (in `src/environments/environment.test.ts`) provides information such as,
+     - Login Credentials (Username).
+     - User Team and Tutorial Information.
+     - "Test Mode" activation by setting `test` property to `true`.
+3. Mock Service Injections
+   - Data reflected in the environment file then assists the application replace the use of some existing services with those that bypass specific functions that are irrelevant to E2E Testing. This includes Authentication Bypassing, Backend API Simulation (so that tests can be carried out in isolationj) and others.
```

Comment:
```suggestion
   - Data reflected in the environment file then assists the application to replace the use of some existing services with those that bypass specific functions that are irrelevant to E2E Testing. This includes Authentication Bypassing, Backend API Simulation (so that tests can be carried out in isolation) and others.
```